lib/ash_phoenix/form/form.ex

```diff
@@ -545,9 +545,17 @@ defmodule AshPhoenix.Form do
     name = opts[:as] || "form"
     id = opts[:id] || opts[:as] || "form"
 
+    params =
+      if is_map(opts[:params]) or is_nil(opts[:params]) do
+        opts[:params] || %{}
+      else
+        IO.warn("The `params` option should be a map.")
+        %{}
```

Comment:
but if someone passed in a string like `"Hello World"` to a field in a form that should be a map, wouldn't you want a friendly error message returned that they passed in a bad value? Similar to passing in a string to an integer field, the validation would fail. A non-null and non-map value for a map field should error out gracefully. I think the very specific case of an empty string `""` could be treated the same as `nil` though. My 2 cents.

```
if is_map(opts[:params]) or is_nil(opts[:params]) or opts[:params] == "" do
```

Reply:
Zach said
>  If we get a non-map for an embedded form we should probably just treat it as if it was an empty map TBH

You advocate that instead the call should return an error?

Reply:
Actually @lardcanoe is correct that in this context `""` should likely be treated as `nil`, which is common in form handling.